src/shared/util.js

```diff
@@ -600,13 +600,14 @@ function string32(value) {
                              (value >> 8) & 0xff, value & 0xff);
 }
 
+// Calculate the base 2 logarithm of the number `x`. This differs from
+// the native function in the sense that it returns the ceiling value
+// and that `Infinity`/`NaN` values are treated as 0.
```

Comment:
This line is really confusing me, since as far as I can tell `log2(Infinity) == Infinity` and `log2(NaN) == NaN` rather than `0`. What's correct here, the code or the comment?

Reply:
Yes, now that I read it again I can see how it's a bit confusing. I rewrote it in the new commit, so hopefully it's better like that.

Reply:
That's a lot clearer, thanks!